music_assistant/providers/audible/audible_helper.py

```diff
@@ -76,85 +107,158 @@ async def get_library(self) -> AsyncGenerator[Audiobook, None]:
-            if page * page_size >= total_items:
-                break
+                try:
+                    if cached_book is not None:
+                        album = await self._parse_audiobook(cached_book)
+                        yield album
+                    else:
+                        album = await self._parse_audiobook(audiobook_data)
+                        yield album
+
+                    total_processed += 1
+                    items_processed_this_page += 1
+                except ValueError as exc:
+                    self.logger.warning(f"Skipping invalid audiobook: {exc}")
+                    total_processed += 1
+                    continue
+
+            self.logger.debug(
+                "Audible: Processed %s valid audiobooks on page %s", items_processed_this_page, page
+            )
 
+            # Continue to the next page
             page += 1
+            self.logger.debug(
+                "Audible: Moving to page %s (processed: %s, total reported: %s)",
+                page,
+                total_processed,
+                total_items,
+            )
+
+        # Log the final results
+        if iteration >= max_iterations:
+            self.logger.warning(
+                "Audible: Reached maximum iteration limit (%s) with %s items processed",
+                max_iterations,
+                total_processed,
+            )
+        else:
+            self.logger.info(
+                "Audible: Successfully retrieved %s audiobooks from library", total_processed
+            )
 
     async def get_audiobook(self, asin: str, use_cache: bool = True) -> Audiobook | None:
         """Fetch the audiobook by asin."""
-        if use_cache:
-            cached_book = await self.mass.cache.get(
-                key=asin, base_key=CACHE_DOMAIN, category=CACHE_CATEGORY_AUDIOBOOK, default=None
+        try:
+            if use_cache:
+                cached_book = await self.mass.cache.get(
+                    key=asin, base_key=CACHE_DOMAIN, category=CACHE_CATEGORY_AUDIOBOOK, default=None
+                )
+                if cached_book is not None:
+                    return await self._parse_audiobook(cached_book)
+            response = await self._call_api(
+                f"library/{asin}",
+                response_groups="""
+                    contributors, media, price, product_attrs, product_desc, product_details,
+                    product_extended_attrs,is_finished
+                    """,
             )
-            if cached_book is not None:
-                return await self._parse_audiobook(cached_book)
-        response = await self._call_api(
-            f"library/{asin}",
-            response_groups="""
-                contributors, media, price, product_attrs, product_desc, product_details,
-                product_extended_attrs,is_finished
-                """,
-        )
 
-        if response is None:
+            if response is None:
+                return None
```

Comment:
you need to raise a MediaNotFoundError here and not return None, that will cause all kind of unexpected results